addons/project/models/project_task.py

```diff
@@ -1684,3 +1689,16 @@ def action_redirect_to_project_task_form(self):
             'url': '/web#model=project.task&id=%s&action=%s&view_type=form' % (self.id, self.env.ref('project.action_view_my_task').id),
             'target': 'new',
         }
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True): #TODO: multiple groupby ??
+        # A read_group can not be performed if records are grouped by personal_stage_type_id as it is a computed field.
+        # personal_stage_type_ids behaves like a M2O from the point of view of the user, we therefore use this field instead.
+        if 'personal_stage_type_id' in groupby and (not lazy or groupby[0] == 'personal_stage_type_id'):
```

Comment:
> it is not always in lazy when the list/kanban views called read_group? 🤔

I think so but does it change something in this case? Here we want to avoid to do the switch m2o/m2m if it is not needed, i.e. when when the content of the groups is not fetched.

Reply:
No indeed, in fact, either we manage multiple group by or we just check if lazy=True. 🙂